src/richie/apps/search/text_indexing.py

```diff
@@ -0,0 +1,149 @@
+"""
+Customize text analysis & mapping to handle multilingual text and manage searches of word
+fragments through n-grams.
+"""
+
+# Dynamic template to match translated strings with their language analyzer
+# NB: expects objects to follow a conventional form, eg:
+#   {
+#       'title': {
+#           'en': '<english title>',
+#           'fr': '<french title>',
+#       },
+#   }
+# Where all the language keys are optional and dynamically added to the mapping at indexing time
+MULTILINGUAL_TEXT = [
+    {
+        "english_strings": {
+            "match_mapping_type": "string",
+            "path_match": "*.en",
+            "mapping": {
+                "type": "text",
+                "fields": {
+                    "language": {"type": "text", "analyzer": "english"},
+                    "trigram": {
+                        "type": "text",
+                        "analyzer": "english_trigram",
+                        # If we apply trigram on the text query, searching "artificial" will
+                        # match "artificial" but also "art" with a lesser score. This is fine
+                        # when results are sorted by score but in our case, sorting is complex
+                        # and grouped in buckets (open, ongoing, coming soon, archived,...). For
+                        # the moment, we consider that using a standard analyzer on the text
+                        # query is good enough.
+                        "search_analyzer": "english",
+                    },
+                },
+            },
+        }
+    },
+    {
+        "french_strings": {
+            "match_mapping_type": "string",
+            "path_match": "*.fr",
+            "mapping": {
+                "type": "text",
+                "fields": {
+                    "language": {"type": "text", "analyzer": "french"},
+                    "trigram": {
+                        "type": "text",
+                        "analyzer": "french_trigram",
+                        # See above note explaining why we don't want to apply trigram on the
+                        # text query.
+                        "search_analyzer": "french",
+                    },
+                },
+            },
+        }
+    },
+]
+
+
+# Settings inspired from
+# https://www.elastic.co/guide/en/elasticsearch/reference/1.7/analysis-lang-analyzer.html
```

Comment:
Maybe point to the current version of the docs? https://www.elastic.co/guide/en/elasticsearch/reference/6.7/analysis-lang-analyzer.html